File: src/RoomBooking.Shared/Entities/Room.cs
using System;
using System.Collections.Generic;
using RoomBooking.Shared.Entities.Abstract;

namespace RoomBooking.Shared.Entities
{
    public class Room : IEntityBase
    {
        public Room()
        {
            RoomReservations = new List<RoomReservation>();
        }

        public string Name { get; set; }
        public RoomType RoomType { get; set; }
        public Hotel Hotel { get; set; }
        public virtual ICollection<RoomReservation> RoomReservations { get; set; }

        public Guid Id { get; set; }
    }

    public enum RoomType
    {
        Single = 1,
        Double = 2,
        Studio = 3,
        Suite = 4
    }
}

File: src/RoomBooking.Server/wwwroot/app/core/domain/room.ts
export class Room {
    Id: string;
    Name: string;
    RoomType: string;
    Hotel: string;
    HotelId: string;

    constructor(id: string,
        name: string,
        roomtype: string,
        hotel: string,
        hotelid: string
    ) {
        this.Id = id;
        this.Name = name;
        this.RoomType = roomtype;
        this.Hotel = hotel;
        this.HotelId = hotelid;
    }
}